### PR TITLE
bug/198_cache_insert_twice_a_user_token_pair

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #198 
* Traces showing logs about the creation in the proxy and `cache-dump.json` file status: 
```
[Sending the first request with pair user:token = pcoello25:1111111111111111]

$ curl -X GET "http://<host>:<port>/webhdfs/v1/user/pcoello25?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: 111111111111111111"

info: pcoello25 is trying to access to /webhdfs/v1/user/pcoello25 with the token 111111111111111111
info: Authentication OK: {"organizations": [], "displayName": "pcoello25", "roles": [], "app_id": "4d1af2eec3754099a4f8dc86bf735068", "isGravatarEnabled": false, "email": "pcoello25mail", "id": "pcoello25"}
info: Authorization OK: user pcoello25 is allowed to access /webhdfs/v1/user/pcoello25
info: Redirecting to http://<host>:<port>

$ cat cache-dump.json 
[{"user":"pcoello25","token":"111111111111111111"}]

==================================================================

[Requested new token and sending the pair user:token = pcoello25:123456123456123456]

$ curl -X GET "http://<host>:<port>/webhdfs/v1/user/pcoello25?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: 123456123456123456"

info: pcoello25 is trying to access to /webhdfs/v1/user/pcoello25 with the token 123456123456123456
info: Authentication OK: {"organizations": [], "displayName": "pcoello25", "roles": [], "app_id": "4d1af2eec3754099a4f8dc86bf735068", "isGravatarEnabled": false, "email": "pcoello25mail", "id": "pcoello25"}
info: Authorization OK: user pcoello25 is allowed to access /webhdfs/v1/user/pcoello25
info: Redirecting to http://<host>:<port>

$ cat cache-dump.json 
[{"user":"pcoello25","token":"123456123456123456"}]
```
* Assignee: @frbattid

### CODE REVIEW

File: cosmos-proxy/src/cache.js [hunk 1]
```diff
@@ -36,17 +36,22 @@ function isCacheEmpty() {
 function isCacheAuthenticated(reqUser, token) {
     if (!isCacheEmpty()) {
         for (var i = 0; i < cache.length; i++) {
-            if ((cache[i]['user'] === reqUser) && (cache[i]['token'] === token)) {
-                return true;
+            if (cache[i].user === reqUser) {
+                if (cache[i].token === token) {
+                    return 1; // 1
```
Comment (frbattid) [inline cosmos-proxy/src/cache.js]: Leftover comment?

File: cosmos-proxy/src/cache.js [hunk 2]
```diff
@@ -36,17 +36,22 @@ function isCacheEmpty() {
...
+                } else {
+                    return 2; // 2
```
Comment (frbattid) [inline cosmos-proxy/src/cache.js]: Leftover comment?

File: cosmos-proxy/src/cache.js [hunk 3]
```diff
@@ -36,17 +36,22 @@ function isCacheEmpty() {
...
-    return false;
+        return 0;
+    } else {
+        return 0;
+    } // if else
 } // isCacheAuthenticated
 
 function updateCacheFile() {
-    fs.writeFileSync(pathToFile,'[' + cache + ']');
+    fs.writeFileSync(pathToFile, JSON.stringify(cache), 'utf-8');
```
Comment (frbattid) [inline cosmos-proxy/src/cache.js]: Ummmm `utf-8` or `utf8`?

File: cosmos-proxy/src/cache.js (hunk 2 above)
Comment (pcoello25) [inline cosmos-proxy/src/cache.js]: Forgot to delete!! 
Fixed in  bcabf88

File: cosmos-proxy/src/cache.js (hunk 1 above)
Comment (pcoello25) [inline cosmos-proxy/src/cache.js]: Forgot to delete!!
Fixed in  bcabf88

File: cosmos-proxy/src/cache.js (hunk 3 above)
Comment (pcoello25) [inline cosmos-proxy/src/cache.js]: It's `utf8`, you're right. Checked in official page https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options :) 
Fixed in  bcabf88

**Comment by frbattid** [comment]
LGTM

**Pull request merged**